packages/eslint-plugin/src/rules/prefer-at.ts

```diff
@@ -0,0 +1,219 @@
+        case AST_NODE_TYPES.PrivateIdentifier:
+          return `#${node.name}`;
+        case AST_NODE_TYPES.Identifier:
+          return node.name;
+        case AST_NODE_TYPES.Literal:
+          return node.raw;
+        case AST_NODE_TYPES.ThisExpression:
+          return 'this';
+        case AST_NODE_TYPES.CallExpression:
+          return `${getFullName(node.callee)}()`;
+        case AST_NODE_TYPES.MemberExpression:
+          if (node.property.type === AST_NODE_TYPES.Literal) {
+            return `${getFullName(node.object)}[${node.property.raw}]`;
+          }
+          return `${getFullName(node.object)}.${getFullName(node.property)}`;
+        default:
+          throw new UnknownNodeError(node);
+      }
+    }
+
+    function hasCallExpression(node: TSESTree.MemberExpression): boolean {
+      return (
+        node.object.type === AST_NODE_TYPES.CallExpression ||
+        (node.object.type === AST_NODE_TYPES.MemberExpression &&
+          hasCallExpression(node.object))
+      );
+    }
+
+    function getTypeAtLocation(node: TSESTree.Node): ts.Type | undefined {
+      const originalNode = parserServices.esTreeNodeToTSNodeMap.get(node);
+      if (!originalNode) {
+        return undefined;
+      }
+      return checker.getTypeAtLocation(originalNode);
+    }
+
+    type SupportedObject = (type: ts.Type) => boolean;
+
+    function checkObjectName(name: string): SupportedObject {
+      return type => type.getSymbol()?.name === name;
+    }
+
+    function checkObjectType(flags: ts.TypeFlags): SupportedObject {
+      return type => type.getFlags() === flags;
+    }
+
+    const supporterObjects: Array<SupportedObject> = [
+      checkObjectName('Array'),
+      checkObjectName('Int8Array'),
+      checkObjectName('Uint8Array'),
+      checkObjectName('Uint8ClampedArray'),
+      checkObjectName('Int16Array'),
+      checkObjectName('Uint16Array'),
+      checkObjectName('Int32Array'),
+      checkObjectName('Float32Array'),
+      checkObjectName('Uint32Array'),
+      checkObjectName('Float64Array'),
+      checkObjectName('BigInt64Array'),
+      checkObjectName('BigUint64Array'),
+      // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+      checkObjectName('String'),
+      checkObjectType(ts.TypeFlags.String),
+    ];
+
+    function isSupportedObject(type: ts.Type): boolean {
+      return supporterObjects.some(check => check(type));
+    }
+
+    function isExpectedObject(
+      node: TSESTree.Node,
+    ): node is TSESTree.MemberExpression {
+      if (
+        node.type !== AST_NODE_TYPES.MemberExpression ||
+        (options.ignoreFunctions && hasCallExpression(node))
+      ) {
+        return false;
+      }
+      const type = getTypeAtLocation(node.object);
+      if (type && !isSupportedObject(type)) {
+        return false;
+      }
+      const atMember = type?.getProperty('at');
+      return (
+        atMember?.declarations?.every(declaration => {
```

Comment:
> `.every`

So if a project adds its own `.at` method augmentation in their own file, this will return false? That's a pity because `@types/node` does that very thing.

[Request] I'm also betting this rule doesn't need to check that the `at` is built-in. Other similar rules such as `prefer-includes` don't. See how that one and other rules that use `getStaticValue` operate.